datafusion/core/tests/dataframe/mod.rs

```diff
@@ -2996,6 +2997,119 @@ async fn test_count_wildcard_on_window() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn reproducer_e2e_with_repartition_sorts_false() -> Result<()> {
+    reproducer_e2e_impl(false).await?;
+
+    // 💥 Doesn't pass, and generates this plan:
+    //
+    // AggregateExec: mode=Final, gby=[id@0 as id], aggr=[], ordering_mode=Sorted
+    //   SortExec: expr=[id@0 ASC NULLS LAST], preserve_partitioning=[false]
+    //     CoalescePartitionsExec
+    //       AggregateExec: mode=Partial, gby=[id@0 as id], aggr=[]
+    //         UnionExec
+    //           DataSourceExec: file_groups={1 group: [[{testdata}/alltypes_tiny_pages.parquet]]}, projection=[id], output_ordering=[id@0 ASC NULLS LAST], file_type=parquet
+    //           DataSourceExec: file_groups={1 group: [[{testdata}/alltypes_tiny_pages.parquet]]}, projection=[id], file_type=parquet
```

Comment:
Since `repartition_sorts`= false. This test fails as expected. I would modify the test to make it pass instead of not pass. See my comment for `reproducer_e2e_impl` below